test/blackbox/common/BlackboxTestsDeadlineQos.cpp

```diff
@@ -72,6 +74,29 @@ class DeadlineQos : public testing::TestWithParam<communication_type>
         }
     }
 
+protected:
+
+    // Convert seconds (double) to Duration_t with seconds + nanoseconds split
+    static eprosima::fastdds::dds::Duration_t seconds_to_duration(
+            double s)
+    {
+        eprosima::fastdds::dds::Duration_t d{};
+        d.seconds = static_cast<int32_t>(s);
+        d.nanosec = static_cast<uint32_t>((s - d.seconds) * 1e9);
+        return d;
+    }
+
+    // Update deadline period on a native DataWriter
+    static void set_deadline_seconds(
+            eprosima::fastdds::dds::DataWriter& native,
+            double seconds)
+    {
+        eprosima::fastdds::dds::DataWriterQos q;
+        ASSERT_EQ(native.get_qos(q), eprosima::fastdds::dds::RETCODE_OK);
+        q.deadline().period = seconds_to_duration(seconds);
+        ASSERT_EQ(native.set_qos(q), eprosima::fastdds::dds::RETCODE_OK);
+    }
```

Comment:
I don't think you need these methods, Time_t already has a constructor taking seconds.